drink/src/session.rs

```diff
@@ -463,6 +463,22 @@ where
         self.call_internal::<_, V>(None, message, args, endowment)
     }
 
+    /// Calls the last deployed contract. Expect it to be reverted and the message result to be of
+    /// type `Result<_, E>`.
+    pub fn call_and_expect_error<S: AsRef<str> + Debug, E: Debug + Decode>(
+        &mut self,
+        message: &str,
+        args: &[S],
+        endowment: Option<BalanceOf<T::Runtime>>,
+    ) -> E {
+        self.call_internal::<_, Result<(), E>>(None, message, args, endowment)
+            .expect_err("Call should fail")
+            .decode_revert::<Result<(), E>>()
+            .expect("Call should be reverted")
```

Comment:
Here we're loosing information - `decode_revert` returns `None` if the underlying error is different than `CallReverted`. So, if it's anything else - we will fail on this expect and not provide any context for the developer to fix it. 

Maybe, instead of returning `Option` from `decode_revert`, we could return a `Result` and on the `Err` side include the original error?

Reply:
done